sysl2/sysl/swagger/types.go

```diff
@@ -0,0 +1,207 @@
+package swagger
+
+import (
+	"fmt"
+	"net/url"
+	"sort"
+	"strings"
+
+	"github.com/go-openapi/spec"
+)
+
+type Type struct {
+	Name       string
+	Properties []Field
+	isAlias    bool
+	isArray    bool
+}
+
+type Field struct {
+	Name     string
+	Type     Type
+	Optional bool
+}
+
+type TypeList []Type
+type FieldList []Field
+
+// nolint:gochecknoglobals
+var builtIns = []string{"int32", "int64", "int", "float", "string", "date", "bool", "decimal", "datetime", "xml"}
+
+// nolint:gochecknoglobals
+var swaggerToSyslMappings = map[string]string{
+	"boolean": "bool",
+	"date":    "date",
+}
+
+// nolint:gochecknoglobals
+var swaggerFormats = []string{"int32", "int64", "float", "double", "date", "date-time", "byte"}
+
+func IsKeyword(name string) bool {
+	for _, kw := range builtIns {
+		if name == kw {
+			return true
+		}
+	}
+	return false
+}
+
+func (t *Type) AddProperties(props map[string]spec.Schema, requiredProps []string, knownTypes *TypeList) {
+	keys := []string{}
+	fields := map[string]Field{}
+	for pname, prop := range props {
+		refType := findReferencedType(prop)
+		propType, _ := knownTypes.Find(refType)
+		if refType == "object" {
+			propType = Type{
+				Name:    fmt.Sprintf("%s_%s_obj", t.Name, pname),
+				isAlias: true,
+			}
+			*knownTypes = append(*knownTypes, propType)
+		}
+		f := Field{
+			Name:     pname,
+			Type:     propType,
+			Optional: !contains(pname, requiredProps),
+		}
+		fields[pname] = f
+		keys = append(keys, pname)
+	}
+
+	sort.SliceStable(keys, func(i, j int) bool {
+		return strings.Compare(keys[i], keys[j]) < 0
+	})
+	for _, k := range keys {
+		t.Properties = append(t.Properties, fields[k])
+	}
+}
+
+func mapSwaggerTypeAndFormatToType(typeName, format string) string {
+	if format != "" && !contains(format, swaggerFormats) {
+		panic("unknown format")
+	}
+
+	conversions := map[string]map[string]string{
+		"string": {
+			"":          "string",
+			"date":      "date",
+			"date-time": "datetime",
+			"byte":      "string",
+		},
+		"integer": {
+			"":      "int",
+			"int32": "int32",
+			"int64": "int64",
+		},
+		"number": {
+			"":       "float",
+			"double": "float",
+			"float":  "float",
+		},
+	}
+
+	if formatMap, ok := conversions[typeName]; ok {
+		if result, ok := formatMap[format]; ok {
+			return result
+		}
+	}
+
```

Comment:
It would be great if a descriptive log is shown for failure scenarios. It would help to find spec issues and potential scenarios that can be handled by importer

Reply:
done